src/states/node.rs

```diff
@@ -654,34 +687,6 @@ impl Node {
         use crate::messages::MessageContent::*;
         use crate::Authority::{Client, ManagedNode, PrefixSection, Section};
 
-        if !self.chain.is_member() {
-            match routing_msg.content {
-                ExpectCandidate { .. }
-                | AcceptAsCandidate { .. }
-                | NeighbourInfo(..)
-                | NeighbourConfirm(..)
-                | Merge(..)
-                | UserMessagePart { .. } => {
-                    // These messages should not be handled before node becomes established
-                    self.add_routing_message_to_backlog(routing_msg);
-                    return Ok(());
-                }
-                ConnectionInfoRequest { .. } => {
-                    if !self.chain.our_prefix().matches(&routing_msg.src.name()) {
-                        self.add_routing_message_to_backlog(routing_msg);
-                        return Ok(());
-                    }
-                }
-                Relocate { .. }
-                | ConnectionInfoResponse { .. }
-                | RelocateResponse { .. }
-                | Ack(..)
-                | NodeApproval { .. } => {
```

Comment:
I think there is a change of behaviour for Establishing:
NodeApproval was handled as normal, thus generate an error, but now it is pushed to the backlog, I think.

```
    fn handle_routing_message(
        &mut self,
        msg: RoutingMessage,
        outbox: &mut EventBox,
    ) -> Result<(), RoutingError> {
        use crate::{messages::MessageContent::*, routing_table::Authority::*};

        let src_name = msg.src.name();

        match msg {
            RoutingMessage {
                content:
                    ConnectionInfoRequest {
                        encrypted_conn_info,
                        pub_id,
                        msg_id,
                    },
                src: ManagedNode(_),
                dst: ManagedNode(_),
            } => {
                if self.our_prefix().matches(&src_name) {
                    self.handle_connection_info_request(
                        encrypted_conn_info,
                        pub_id,
                        msg_id,
                        msg.src,
                        msg.dst,
                        outbox,
                    )
                } else {
                    self.add_message_to_backlog(RoutingMessage {
                        content: ConnectionInfoRequest {
                            encrypted_conn_info,
                            pub_id,
                            msg_id,
                        },
                        ..msg
                    });
                    Ok(())
                }
            }
            RoutingMessage {
                content:
                    ConnectionInfoResponse {
                        encrypted_conn_info,
                        pub_id,
                        msg_id,
                    },
                src: ManagedNode(src_name),
                dst: Client { .. },
            } => self.handle_connection_info_response(
                encrypted_conn_info,
                pub_id,
                msg_id,
                src_name,
                msg.dst,
            ),
            RoutingMessage {
                content: Ack(ack, _),
                ..
            } => {
                self.handle_ack_response(ack);
                Ok(())
            }
            _ => {
                self.add_message_to_backlog(msg);
                Ok(())
            }
        }
    }
```

Reply:
`NodeApproval` is a message sent to `ProvingNode` to inform it that it passed the resource proof. It doesn't make sense to send it to `EstablishingNode` or `Node`, so it's ignored there (with a log message). I don't think it matters much whether we log the error in `EstablishingNode` already, or just backlog it and log it once we are `Node`. It doesn't affect anything and it makes the code simpler. So even though strictly speaking it is a behaviour change, it's a very minor one and I think it's justified by making the code sightly simpler.

Reply:
Ideally Refactoring commit does not change any behaviour.
With such a big change, it is difficult to know what was intended change in behaviour and unintended ones (difficult enough to detect if there are any change). As a result review need to flag all of them and assume all are unintended.

Best to document all these kind of changes so we know it is intentional, and explain why they are sensible, and keep them to a minimum / separate commit if sensible.

Reply:
> Best to document all these kind of changes so we know it is intentional, and explain why they are sensible, and keep them to a minimum / separate commit if sensible.

Reply:
> Best to document all these kind of changes so we know it is intentional, and explain why they are sensible, and keep them to a minimum / separate commit if sensible.

Maybe add it to the PR description, which probably could be fleshed out a bit.